lib/active_fedora/associations/collection_association.rb

```diff
@@ -367,7 +367,11 @@ def delete_or_destroy(records, method)
 
           records.each { |record| callback(:before_remove, record) }
           delete_records(existing_records, method) if existing_records.any?
-          records.each do |record|
+          # Existing_records are deleted in the line above, therefore
+          # I am excluding them from the loop below. This seems to be
+          # causing problems with 4 integration tests. Still need to
+          # figure out why.
+          (records-existing_records).each do |record|
```

Comment:
I figured out the rational for this code. The call to `delete_records()` issues the delete to Fedora/Solr. The call to `target.delete()` removes the element from the array/collection.